gnuradio-runtime/python/gnuradio/gr/bindings/block_gateway_python.cc

```diff
@@ -7,17 +7,200 @@
+        return tags;
+    }
+
+    /*******************************************************************
+     * Overloads for various scheduler-called functions
+     ******************************************************************/
+    void forecast(int noutput_items, gr_vector_int& ninput_items_required) override
+    {
+        py::gil_scoped_acquire acquire;
+
+        py::object ret_ninput_items_required = _py_handle.attr("handle_forecast")(
+            noutput_items, ninput_items_required.size());
+        ninput_items_required = ret_ninput_items_required.cast<std::vector<int>>();
+    }
+
+    int general_work(int noutput_items,
+                     gr_vector_int& ninput_items,
+                     gr_vector_const_void_star& input_items,
+                     gr_vector_void_star& output_items) override
+    {
+        py::gil_scoped_acquire acquire;
+
+        py::object ret = _py_handle.attr("handle_general_work")(
+            noutput_items, ninput_items, input_items, output_items);
+
+        return ret.cast<int>();
+        ;
+    }
+
+    bool start(void) override
+    {
+        py::gil_scoped_acquire acquire;
+
+        py::object ret = _py_handle.attr("start")();
+        return ret.cast<bool>();
+    }
+
+    bool stop(void) override
+    {
+        py::gil_scoped_acquire acquire;
+
+        py::object ret = _py_handle.attr("stop")();
+        return ret.cast<bool>();
+    }
+
+    void set_msg_handler_pybind(pmt::pmt_t which_port, std::string& handler_name)
+    {
+        if (msg_queue.find(which_port) == msg_queue.end()) {
+            throw std::runtime_error(
+                "attempt to set_msg_handler_pybind() on invalid input message port!");
+        }
+        d_msg_handlers_pybind[which_port] = handler_name;
+    }
+
+protected:
+    // Message handlers back into python using pybind API
+    typedef std::map<pmt::pmt_t, std::string, pmt::comparator> msg_handlers_pybind_t;
+    msg_handlers_pybind_t d_msg_handlers_pybind;
+
+    bool has_msg_handler(pmt::pmt_t which_port) override
+    {
+        if (d_msg_handlers_pybind.find(which_port) != d_msg_handlers_pybind.end()) {
+            return true;
+        } else {
+            return gr::basic_block::has_msg_handler(which_port);
+        }
+    }
+
+    void dispatch_msg(pmt::pmt_t which_port, pmt::pmt_t msg) override
+    {
+        // Is there a handler?
+        if (d_msg_handlers_pybind.find(which_port) != d_msg_handlers_pybind.end()) {
+            // d_msg_handlers_pybind[which_port]->calleval(msg); // Yes, invoke it.
+            py::gil_scoped_acquire acquire;
+            // std::string handler_name(d_msg_handlers_pybind[which_port]);
+            py::object ret =
+                _py_handle.attr(d_msg_handlers_pybind[which_port].c_str())(msg);
+        } else {
+            // Pass to generic dispatcher if not found
+            gr::basic_block::dispatch_msg(which_port, msg);
```

Comment:
just curious if this method is defined in `gr::block` and you want to skip that version, and go to the grandparent implementation, or if there's any other reason to scope it so explicitly?

(yes, I realize this code was fork-lifted, but it's an opportunity for re-examination).

My preferred pattern (which you may disagree with) is
```c++
class Foo : public Bar
{
    using Base = Bar;
    ...
    void baz() override
    {
        Base::baz();
    }
};
```